x-pack/plugins/apm/typings/es_schemas/Transaction.ts

```diff
@@ -56,6 +56,16 @@ export interface TransactionV1 extends APMDocV1 {
     };
     type: string;
   };
+  kubernetes?: {
+    pod: {
+      uid: string;
+    };
+  };
+  docker?: {
+    container: {
+      id: string;
+    };
+  };
```

Comment:
V1 docs are only docs ingested by apm-server pre 6.5, so I don't think those will have `kubernetes` or `docker`. Can you double check with the apm-server team?

Reply:
if that's the case, i'll also have to change the idx call to something like this to avoid type errors:

```
const containerId = idx(transaction as TransactionV2, _ => _.docker.container.id);
```

Reply:
In this case you are casting `transaction` and thereby reproving the compiler. But isn't it actually right? How do you know that it's a `TransactionV2`? Couldn't it be a `TransactionV1`?